runtime/lua/vim/treesitter/query.lua

```diff
@@ -34,14 +72,36 @@ local function safe_read(filename, read_quantifier)
   return content
 end
 
+--- Disables loading query files for {lang} matching provided {patterns}.
+---
+--- If {patterns} is nil, it clears the ignore patterns.
+--- If {patterns} is string[], it replaces the currently set patterns (if any).
+---
+--- You can also pass a function as {patterns}, which will accept a table containing
+--- currently set patterns. You can modify that table and return a new table or nil.
+--- The return value will be used as the new ignore patterns.
+---
+---@param lang The language
+---@param patterns Ignore patterns or callback function
+function M.ignore_query_files(lang, patterns)
```

Comment:
why is this in treesitter/query.lua instead of toplevel `treesitter.lua`? I am sure this is only the start of the configuration requests, and I could imagine more configuration related to languages, highlighting--not only queries.

Reply:
Added this here because the `set_query` function was in this file. I'm open to changing it, just point me towards where it should go and how it should be implemented. 😃